src/state/farms/fetchPublicFarmData.ts

```diff
@@ -15,6 +16,64 @@ type PublicFarmData = {
   multiplier: string
 }
 
+const fetchFarmCalls = (farm: SerializedFarm) => {
+  const { lpAddresses, token, quoteToken } = farm
+  const lpAddress = getAddress(lpAddresses)
+  return [
+    // Balance of token in the LP contract
+    {
+      address: token.address,
+      name: 'balanceOf',
+      params: [lpAddress],
+    },
+    // Balance of quote token on LP contract
+    {
+      address: quoteToken.address,
+      name: 'balanceOf',
+      params: [lpAddress],
+    },
+    // Balance of LP tokens in the master chef contract
+    {
+      address: lpAddress,
+      name: 'balanceOf',
+      params: [getMasterChefAddress()],
+    },
+    // Total supply of LP tokens
+    {
+      address: lpAddress,
+      name: 'totalSupply',
+    },
+    // Token decimals
+    {
+      address: token.address,
+      name: 'decimals',
+    },
+    // Quote token decimals
+    {
+      address: quoteToken.address,
+      name: 'decimals',
+    },
+  ]
+}
+
+export const fetchPublicFarmsData = async (farms: SerializedFarmConfig[]): Promise<any[]> => {
+  const farmCalls = farms.flatMap((farm) => fetchFarmCalls(farm))
+  const chunkSize = farmCalls.length / farms.length
+  const farmMultiCallResult = await multicallv2(erc20, farmCalls)
+  return farmMultiCallResult.reduce((resultArray, item, index) => {
+    const chunkIndex = Math.floor(index / chunkSize)
+
+    if (!resultArray[chunkIndex]) {
+      // eslint-disable-next-line no-param-reassign
+      resultArray[chunkIndex] = [] // start a new chunk
+    }
+
+    resultArray[chunkIndex].push(item)
+
+    return resultArray
+  }, [])
+}
+
 const fetchFarm = async (farm: SerializedFarm): Promise<PublicFarmData> => {
```

Comment:
this `fetchFarm` was only used in `fetchFarm.ts` which is now removed. Save to remove 🕺

Reply:
Done